### ISSUE TITLE
FIltrating predicted mutation names according to a specific pattern?

### ISSUE THREAD

**Issue by sunbrn** [armando2603/virus-project#17]
I analyzed the paper

and got several predictions (and this is ok). 
<img width="1379" alt="image" src="https://user-images.githubusercontent.com/11488353/162974150-8ee0f865-86a5-4175-a90a-bfb1c9a101e2.png">

Looking at the last one:
SPIKE_S477N+SPIKE_E

Clearly, the second mutation has a "wrong format". I wonder if it is possible to avoid making predictions where the single mutation (or mutation inside a group) have a bad format. A syntactic check (maybe regex-based) could be introduced to say:

^[a-zA-Z0-9]+\_[a-zA-Z\-]+[0-9]+[a-zA-Z\-]+$

in other words: the underscore must always be preceded by a proper protein name and followed by a proper mutation signature (i.e., some letters or dash, a number, and other letters or dash)

**Comment by armando2603** [comment]
Yes it can be done, we have to define proper rule for every cases (single mutation, group and variant).
It can be also that the model stop making errors during time deleting/editing the error during the online learning.

**Comment by sunbrn** [comment]
ok so we leave this for future work, if the performances are still bad after proper online training
